sdk/src/types/block/address/mod.rs

```diff
@@ -71,6 +71,36 @@ impl core::fmt::Debug for Address {
     }
 }
 
+macro_rules! def_is_as_opt {
```

Comment:
This can be done in more places than addresses

Reply:
yeah, just figured I'd test the water here before going all out